main.go

```diff
@@ -70,39 +72,63 @@ type APIHelpers interface {
 	UpdateNode(*k8sclient.Clientset, *api.Node) error
 }
 
+// Command line arguments
+type Args struct {
+	labelWhiteList string
+	noPublish      bool
+	oneshot        bool
+	sleepInterval  int
+	sources        []string
+}
+
 func main() {
 	// Assert that the version is known
 	if version == "" {
 		stderrLogger.Fatalf("main.version not set! Set -ldflags \"-X main.version `git describe --tags --dirty --always`\" during build or run.")
 	}
 
 	// Parse command-line arguments.
-	noPublish, sourcesArg, whiteListArg := argsParse(nil)
+	args := argsParse(nil)
 
 	// Configure the parameters for feature discovery.
-	sources, labelWhiteList, err := configureParameters(sourcesArg, whiteListArg)
+	enabledSources, labelWhiteList, err := configureParameters(args.sources, args.labelWhiteList)
 	if err != nil {
 		stderrLogger.Fatalf("error occurred while configuring parameters: %s", err.Error())
 	}
 
-	// Get the set of feature labels.
-	labels := createFeatureLabels(sources, labelWhiteList)
-
 	helper := APIHelpers(k8sHelpers{})
-	// Update the node with the feature labels.
-	err = updateNodeWithFeatureLabels(helper, noPublish, labels)
-	if err != nil {
-		stderrLogger.Fatalf("error occurred while updating node with feature labels: %s", err.Error())
+
+	for {
+		// Get the set of feature labels.
+		labels := createFeatureLabels(enabledSources, labelWhiteList)
+
+		// Update the node with the feature labels.
+		err = updateNodeWithFeatureLabels(helper, args.noPublish, labels)
+		if err != nil {
+			stderrLogger.Fatalf("error occurred while updating node with feature labels: %s", err.Error())
+		}
+
+		if args.oneshot == true {
+			break
+		}
+
+		if args.sleepInterval > 0 {
```

Comment:
I don't we should allow too low values for sleep interval. If the value is too low, the DaemonSet pods might go into a crash loop back off state. WDYT?

Reply:
As I commented earlier, it makes sense to restrict the sleep interval to some sane value. However, I would change it so that too small but non-zero values are forced to the minimum and a warning is printed. Zero value is used to specify an infinite sleep (i.e. no re-labeling).